### PR TITLE
TF Model train and eval step metrics for seq2seq models.

### PR DESCRIPTION
**Pull request by pedro-r-marques**
When using a model with a seq2seq output compute metrics against logits.

# What does this PR do?

This PR changes the TF train and test steps so that metrics can be correctly computed when using keras model.fit.

The keras Model train/test step functions are supposed to compare the labels (y_true) and the predictions (y_pred).
The previous code was passing as y_pred the ModelOutput dataclass (basically a dict) of values which results in
TF/keras attempting to compute metrics between the variable 'y' (y_true) and each of the elements in the ModelOutput dict.

## Who can review?

@sgugger
@Rocketknight1

### CODE REVIEW

**Comment by Rocketknight1** [comment]
This seems like a good fix, thank you! The changes to `train_step` are very recent and we were expecting some initial problems like this. I'm going to do some testing with your PR today/tomorrow and hopefully approve and merge after that.

**Review by Rocketknight1 — APPROVED** [review]

**Comment by pedro-r-marques** [comment]
@Rocketknight1 I did some more testing. The patch I proposed solves one issue which is that the metrics function no longer gets called against the returned 'loss' value and 'past_key_values'; unfortunately it doesn't quite work correctly in the sense that the metrics are not correctly returned. Internally the model.metrics still has 5 metric variables 2 for the loss (a.k.a loss and loss_loss) and 3 for the output values... only one of which is really valid. I need to dig deeper... create a unit test and resubmit when I've a more complete solution.

**Comment by pedro-r-marques** [comment]
@Rocketknight1 apologies for squashing your changes unintentionally :-(. Cleaned up the git log; hopefully preserving your changes this time and took another roll at the CI dice.

**Comment by Rocketknight1** [comment]
@pedro-r-marques It's okay, I wrecked your changes too. Git is just hard, lol. Anyway, it looks good now and tests are green, so merging!
